01_introduction_to_algorithms/c/01_binary_search.c

```diff
@@ -0,0 +1,39 @@
+#include <stdio.h>
+#include <stdlib.h>
+
+#define SIZE 4;
+
+int binarySearch(int[], int);
+
+int main()
+{
+    int myList[] = {1, 3, 5, 7, 9};
+    printf("%d\n", binarySearch(myList, 3));  // 1
+    printf("%d\n", binarySearch(myList, -1)); //-1
+    return 0;
+}
+
+int binarySearch(int list[], int item)
+{
+    int low = 0;
+    int high = SIZE;
```

Comment:
Instead of using a `SIZE` constant, can you do:

```
int high = sizeof(list)/sizeof(list[0]);
```

That way this code will work with arrays of any size

Reply:
If you want to do that I have to calculate the number of arrays directly in the main program, add a len parameter to the function.